tooling/nargo_fmt/src/utils.rs

```diff
@@ -251,15 +251,19 @@ impl Item for Param {
     }
 
     fn format(self, visitor: &FmtVisitor, shape: Shape) -> String {
+        let pattern = visitor.slice(self.pattern.span());
         let visibility = match self.visibility {
             Visibility::Public => "pub ",
             Visibility::Private => "",
             Visibility::DataBus => "call_data",
         };
-        let pattern = visitor.slice(self.pattern.span());
-        let ty = rewrite::typ(visitor, shape, self.typ);
 
-        format!("{pattern}: {visibility}{ty}")
+        if self.typ.synthesized {
+            pattern.to_string()
+        } else {
+            let ty = rewrite::typ(visitor, shape, self.typ);
+            format!("{pattern}: {visibility}{ty}")
+        }
```

Comment:
Instead of having this synthesized field on types, I think it'd be cleaner to have it on the pattern itself. It looks like it's currently only used for `self` so maybe we can have a flag for that specifically.

We'll still have to handle `&mut self` as well.